pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -526,6 +529,32 @@ private void handleUpsert(ImmutableSegment immutableSegment) {
             _tableNameWithType));
     PartitionUpsertMetadataManager partitionUpsertMetadataManager =
         _tableUpsertMetadataManager.getOrCreatePartitionManager(partitionId);
+    return partitionUpsertMetadataManager;
+  }
+
+  /**
+   * Handle Upsert TTL during segment state change from CONSUMING to ONLINE.
+   *
+   * When a new segments commit, we will remove primary key indexes from heap that has timestamp < (eventTime - TTL).
```

Comment:
change to "when a new segment commits"

Reply:
Heap is too general. Change the term heap to the exact data structure used

Reply:
thanks, updated.